### PR TITLE
#16115 reverting changes introduced in #7918

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotmarketing/business/IdentifierConsistencyIntegrationTest.java [hunk 1]
```diff
@@ -85,23 +88,23 @@ public void Test_Identifier_Create_Folder_Tree_Then_Update_Update_Identifier_Non
         final FolderAPI folderAPI = APILocator.getFolderAPI();
         final IdentifierAPI identifierAPI = APILocator.getIdentifierAPI();
         final User user = APILocator.systemUser();
-        final Host host = APILocator.getHostAPI().findByName("demo.dotcms.com", user, false);
+        final Host host = APILocator.getHostAPI().findByName(DEMO, user, false);
```
[the remark this answers is not in the document]
Comment (fabrizzio-dotCMS) [reply dotCMS/src/integration-test/java/com/dotmarketing/business/IdentifierConsistencyIntegrationTest.java]: ok np.

File: dotCMS/src/integration-test/java/com/dotmarketing/business/IdentifierConsistencyIntegrationTest.java [hunk 2]
```diff
@@ -85,23 +88,23 @@ public void Test_Identifier_Create_Folder_Tree_Then_Update_Update_Identifier_Non
...
         final String rootFolderName = String.format("/%sroot2", prefix);
         Folder root = null;
 
         root = folderAPI.createFolders(rootFolderName, host, user, false);
 
-        final Folder f1 = folderAPI
+        final Folder folder1 = folderAPI
                 .createFolders(root.getPath() + "child-1/child-1-1/child-1-1-1/child-1-1-1-1",
```
[the remark this answers is not in the document]
Comment (fabrizzio-dotCMS) [reply dotCMS/src/integration-test/java/com/dotmarketing/business/IdentifierConsistencyIntegrationTest.java]: what do you have in mind?

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05070AddIdentifierVirtualColumn.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    private static final String MS_SQL_CHECK_VIRTUAL_COLUMN = "SELECT * FROM INFORMATION_SCHEMA.COLUMNS WHERE TABLE_NAME = 'identifier' AND COLUMN_NAME = 'full_path_lc'";
+
+    private static final String MSSQL_ADD_VIRTUAL_COLUMN = "ALTER TABLE identifier ADD full_path_lc  as CASE WHEN parent_path = 'System Host' THEN '/' ELSE LOWER(CONCAT(parent_path, asset_name)) END";
+    private static final String ORACLE_ADD_VIRTUAL_COLUMN = "ALTER TABLE identifier ADD full_path_lc as ( CASE WHEN parent_path = 'System folder' THEN '/' ELSE  lower(concat(parent_path, asset_name)) END)";
+    private static final String POSTGRES_CREATE_CREATE_VIRTUAL_COLUMN_FUNCTION =
+            "CREATE OR REPLACE FUNCTION full_path_lc(identifier) RETURNS text\n"
+                    + "    AS ' SELECT CASE WHEN $1.parent_path = ''/System folder'' then ''/'' else LOWER($1.parent_path || $1.asset_name) end; '\n"
+                    + "LANGUAGE SQL;\n";
+    private static final String MY_SQL_ADD_VIRTUAL_COLUMN = "ALTER TABLE identifier add column full_path_lc varchar(510) as ( IF(parent_path = 'System Folder', '/', concat(parent_path, asset_name)))";
```
Comment (jgambarios) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05070AddIdentifierVirtualColumn.java]: I think we are missing a `lower` before the `concat(parent_path, asset_name)`

File: dotCMS/src/main/resources/mysql.sql [hunk 4]
```diff
@@ -1037,6 +1037,7 @@ create table identifier (
    asset_type varchar(64),
    syspublish_date datetime,
    sysexpire_date datetime,
+   full_path_lc varchar(510) as ( IF(parent_path = 'System folder', '/', concat(parent_path, asset_name))),
```
Comment (jgambarios) [inline dotCMS/src/main/resources/mysql.sql]: We are also missing a lower here

**Review by jgambarios — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotmarketing/business/IdentifierFactoryImpl.java [hunk 5]
```diff
@@ -176,18 +170,24 @@ protected Identifier findByURI(final String siteId, String uri) throws DotDataEx
...
+           return Collections.emptyList();
+        }
+        parentPath = identifier.getURI();
+
+	    if(!parentPath.endsWith("/")) {
+	        parentPath=parentPath+"/";
 	    }
-	    parent_path = parent_path.toLowerCase();
 
+        Logger.info(getClass(),parentPath);
```
[the remark this answers is not in the document]
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotmarketing/business/IdentifierFactoryImpl.java]: done

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05070AddIdentifierVirtualColumn.java (hunk 3 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task05070AddIdentifierVirtualColumn.java]: done

File: dotCMS/src/main/resources/mysql.sql (hunk 4 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/resources/mysql.sql]: done

File: dotCMS/src/main/java/com/dotcms/integritycheckers/ContentPageIntegrityChecker.java [hunk 6]
```diff
@@ -218,8 +218,7 @@ private void checkPages(final String endpointId, IntegrityType type) throws IOEx
 				dc.addParam(workingInode);
 				dc.addParam(liveInode);
 				dc.addParam(htmlPageIdentifier);
-				dc.addParam(htmlPageParentPath);
-				dc.addParam(htmlPageAssetName);
+				dc.addParam((htmlPageParentPath + htmlPageAssetName).toLowerCase());
```
Comment (jgambarios) [inline dotCMS/src/main/java/com/dotcms/integritycheckers/ContentPageIntegrityChecker.java]: Do we need the `toLowerCase` in line 212?
```
final String htmlPageAssetName = getStringIfNotBlank(
						"asset_name", htmlpages.get(4).toLowerCase());
```

**Review by jgambarios — APPROVED** [review]

**Pull request closed without merging**